### PR TITLE
LocalBytesStreamer

### PR DESCRIPTION
**Pull request by jbouffard**
This is a local implementation of BytesStreamer that allows for users to stream in local files.

### CODE REVIEW

File: util/src/main/scala/geotrellis/util/LocalBytesStreamer.scala [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+ * @return A new instance of LocalBytesStreamer
+ */
+class LocalBytesStreamer(path: String, val chunkSize: Int) extends BytesStreamer {
+  private val f: File = new File(path)
+  private val inputStream: FileInputStream = new FileInputStream(f)
+  private val channel: FileChannel =  inputStream.getChannel
+
+  def objectLength: Long = channel.size
+
+  def getArray(start: Long, length: Long): Array[Byte] = {
```
Comment (jbouffard) [inline util/src/main/scala/geotrellis/util/LocalBytesStreamer.scala]: This is just a modified version of [slurp](https://github.com/geotrellis/geotrellis/blob/master/util/src/main/scala/geotrellis/util/Filesystem.scala#L52). Unfortunately, I wasn't able to find another way of doing ranged, byte reads from a local file stream.

**Comment by jbouffard** [comment]
@lossyrob This is ready for review.

File: raster-test/src/test/scala/geotrellis/raster/io/geotiff/LocalBytesStreamSpec.scala [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+class LocalBytesStreamerSpec extends FunSpec
+  with Matchers
+  with GeoTiffTestUtils {
+
+  describe("Streaming bytes locally") {
+    val path = geoTiffPath("ls8_int32.tif")
+    val geoTiffBytes = Files.readAllBytes(Paths.get(path))
+    val buffer = ByteBuffer.wrap(geoTiffBytes)
+    val chunkSize = 2000
+    val local = new MockLocalArrayBytes(chunkSize, geoTiffBytes)
```
Comment (lossyrob) [inline raster-test/src/test/scala/geotrellis/raster/io/geotiff/LocalBytesStreamSpec.scala]: We read directly off the filesystem for other tests, why make a mock here?

Comment (jbouffard) [inline raster-test/src/test/scala/geotrellis/raster/io/geotiff/LocalBytesStreamSpec.scala]: I guess that didn't really occur to me when I was writing it... I got rid of it, though.

**Pull request merged**